src/runtime_src/hip/core/memory.h

```diff
@@ -145,10 +145,10 @@ namespace xrt::core::hip
     remove(uint64_t addr);
   
     std::shared_ptr<xrt::core::hip::memory>
-    get_hip_mem_from_addr(void* addr);
+    get_hip_mem_from_addr(void* addr, size_t* offset=nullptr);
```

Comment:
Overall looks good, 
One last comment. Returning values in arguments is kind of a C style coding.
Please change the return value to std::pair<std::shared_ptr<memory, uint64_t> . send hip::memory and offset in that pair. Use it all the places where ever we are using

Reply:
fixed.